plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -66,9 +66,11 @@ def set_deployment_metadata
       end
     end
 
+    # labels that are used to match the previous deployment,
+    # they cannot change or we will create a new deployment instead of updating the previous one
+    # re-creating roles or re-creating the deploy group will lead to duplicate deployments
     def deployment_labels
-      # Deployment labels should not include the ids of the release, role or deploy groups
-      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
+      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
```

Comment:
Sorry, was coming home.

Not sure I understand, what is to "keep the nice names in the Deployment labels"?

I think this is used to match the pods to replace only (as the previous comment also says). Without this change (or technically, the one I did that is very similar), when creating a new release, the old pods continue to be there running. Looking quickly at the file, it seems because it is used in set_selector_metadata() (like the comment in that func says, it is used for matching).

The tag, revision, etc. labels are in the pods. I confirm I run this and the labels were there after a new deployment. This change does not remove those labels from the pods, if I tested correctly.